src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,301 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2010 Alex Buloichik
+               Home page: http://www.omegat.org/
+               Support center: http://groups.yahoo.com/group/OmegaT/
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.machinetranslators;
+import java.awt.Window;
+import java.io.BufferedReader;
+import java.io.InputStreamReader; 
+import java.net.MalformedURLException;
+import java.util.Map;  
+import org.omegat.gui.exttrans.MTConfigDialog;
+import org.omegat.util.JsonParser;
+import org.omegat.util.Language;
+import org.omegat.util.Log;
+import org.omegat.util.OStrings;   
+import org.omegat.util.Preferences;
+import org.omegat.util.StringUtil;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.OutputStream;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+/**
+ *
+ * @author prompsit
+ */
+
+class MTHUBTranslateRequest{
```

Comment:
Before I do a more substantive review, can you clean up all the whitespace errors? If you're not sure where the problems are, run `./gradlew checkstyleMain` and look at the report at `./build/reports/checkstyle/main.html`. If you are running e.g. Eclipse and have the Checkstyle plugin installed, you may even be able to automate the fixes.

Reply:
ok, I'm on it.

Reply:
Done!